config/routes.rb

```diff
@@ -709,6 +709,7 @@
     get '/tasks' => 'tasks#index', as: :tasks
     post '/tasks/create-fake-provider' => 'tasks#create_fake_provider'
     post '/tasks/:task' => 'tasks#run', as: :run_task
+    get '/tasks/:task/confirm' => 'tasks#confirm', as: :confirm_task
```

Comment:
Would it be simpler to have  an explicit path for this? Then no need for a controller action at all, I think, and 404 would work out of the box?